crates/common/src/compile.rs

```diff
@@ -271,9 +271,41 @@ impl ProjectCompiler {
     }
 }
 
-/// Map over artifacts contract sources name -> file_id -> (source, contract)
+/// Contract source code and bytecode.
 #[derive(Clone, Debug, Default)]
-pub struct ContractSources(pub HashMap<String, HashMap<u32, (String, ContractBytecodeSome)>>);
+pub struct ContractSources {
+    /// Map over artifacts contract sources name -> file_id -> (source, contract)
+    pub sources_by_name: HashMap<String, HashMap<u32, (String, ContractBytecodeSome)>>,
```

Comment:
Can this be `String -> Vec<u32>` to not duplicate `sources_by_id`?

Reply:
Good call!